app/models/cluster_file_info.rb

```diff
@@ -3,24 +3,36 @@ class ClusterFileInfo
 
   embedded_in :study_file
 
-  field :custom_colors, type: Hash, default: {}
+  field :custom_colors, type: String
+  # hash of annotation names to booleans of whether the annotation should, by default, be shown split
+  # on pipes (assuming it is array-based).  Unspecified is treated as false by the front end
+  field :annotation_split_defaults, type: String
 
-  COLOR_ENCODE_TRANSFORM = %i[encode64 decode64]
+  before_save :stringify_hashes
 
-  # decode base64 annotation/label names back into human-readable text
-  def decoded_custom_colors
-    self.class.transform_custom_colors(custom_colors, :decode64)
+  # decode string back to Hash
+  def custom_colors_as_hash
+    ActiveSupport::JSON.decode(custom_colors || '{}')
   end
 
-  # helper to transform a nested custom color hash to/from base64
-  def self.transform_custom_colors(hash, transform = :encode64)
-    raise ArgumentError, "\"#{transform}]\" is not a valid transform" unless COLOR_ENCODE_TRANSFORM.include?(transform)
+   # decode string back to Hash
+   def annotation_split_defaults_as_hash
+    ActiveSupport::JSON.decode(annotation_split_defaults || '{}')
+  end
 
-    hash.map do |key, nested_hash|
-      {
-        Base64.send(transform, key) => nested_hash.map { |k, v| { Base64.send(transform, k) => v } }.reduce({}, :merge)
-      }
-    end.reduce({}, :merge)
+  # This is called before the data is saved to the DB to ensure the file hashes are encoded as JSON strings
+  # This is necessary to work around MongoDBs constraints on '.' and '$' in hash keys
+  def stringify_hashes
```

Comment:
Very slick - neatly sidesteps the original issue, and maintains a "human readable" form in the DB.